addon/bootstrap.js

```diff
@@ -58,6 +58,11 @@ function shouldDisable() {
 }
 
 function handleStartup() {
+  // If the preference observer fires before startup completes, the URI won't
+  // be available yet, so just bail until startup has run.
+  if (!addonResourceURI) {
```

Comment:
This isn't going to work, the error we're getting is that `addonResourceURI.QueryInterface(Ci.nsIFileURL)` is throwing `NS_NOINTERFACE` - so it can't be converted to an `nsIFileURL`.

I'm starting to look at the options here.